Automattic/wp-calypso
# Backup: Use feature check rather than custom selector

#### Changes proposed in this Pull Request

* Updates the Backup controller to use a feature check rather than plan check.
* Removes `siteHasSubscription` selector and its tests.

#### Testing instructions





1. Checkout and run this PR in both Calypso blue and green concurrently.
2. [Create a JN Multisite install](https://jurassic.ninja/specialops/) and connect Jetpack. Just Jetpack "Free", for now.
3. Go to https://container-priceless-stonebraker.calypso.live/backup, select your multisite,  and verify you see the "WordPress multi-sites are not supported." backup status card.
4. Using the Store Admin, add a Jetpack Backup subscription to the site (or any Plan that includes Backup).  Backup Realtime product or plan is preferred because then you'll be able to view the "More backups from this day" ActivityCard component containing the additional Download/Restore action buttons.
5. Now that the site has an active Backup subscription, go back to https://container-priceless-stonebraker.calypso.live/backup and select your multisite.
6. Verify you see the Backup Status page like the screenshot below.
7. Verify the Multisite notice/message shows and looks like the screenshot.
8. Verify the "Download backup" button is showing and looks like screenshot.
9. Verify the "Restore to this point" button does not show. (as shown in screenshot)

![Markup 2021-05-25 at 16 55 17](https://user-images.githubusercontent.com/11078128/119567297-26d3b300-bd7a-11eb-8c7e-0fd134be0c79.png)

Related to p4TIVU-a66-p2
